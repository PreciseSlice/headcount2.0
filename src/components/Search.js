import React from 'react';
import '../styles/Search.css';
import PropTypes from 'prop-types';

const Search = props => {
  return (
    <header>
      <h5>
        Click two Colorado school districts to compare their full-day
        kindergarten participation rates
      </h5>
      <form>
        <label htmlFor="search-input">Search for your school district: </label>
        <input
          id="search-input"
          type="text"
          onChange={event => props.searchDistrict(event.target.value)}
          onKeyPress={event => {
            if (event.key === 'Enter') event.preventDefault();
          }}
        />
        <button
          className="clear-button"
          onClick={event => props.handleButtonClick}
        >
          Clear All
        </button>
      </form>
    </header>
  );
};

Search.propTypes = {
  searchDistrict: PropTypes.func.isRequired,
  handleButtonClick: PropTypes.func.isRequired
};

export default Search;
