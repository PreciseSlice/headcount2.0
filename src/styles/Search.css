form {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  padding-bottom: 20px
}

label {
  color: white;
  font-weight: 700;
}

input {
  margin: 10px;
  width: 220px;
}

h5 {
  font-size: 1rem;
  font-weight: 400;
  margin: 5px 0px;
}

button {
  background-color: #5B5BE8;
  border: 1px solid #5B5BE8;
  color: white;
  font-family: 'Open Sans', sans-serif;
  font-weight: 700;
}