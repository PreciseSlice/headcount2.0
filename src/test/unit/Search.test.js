/* eslint-disable */
import React from 'react';
import { shallow, mount, render } from 'enzyme';
import Search from '../../components/Search';
import { containerData, cardData } from './mockData';

const searchDistrict = function() {
  return containerData;
};

describe('Search', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(<Search searchDistrict={searchDistrict} />);
  });

  it('exists', () => {
    expect(wrapper).toBeDefined();
  });

  it('should start out with a state with display set to an empty string', () => {
    expect(wrapper.state().display).toEqual('');
  });

  it('should have a label, input field, and a button in a form element', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should take a search input and set state.display to that input', () => {
    const event = {
      preventDefault() {},
      target: { value: 'colorado' }
    };
    expect(wrapper.state().display).toEqual('');
    expect(wrapper.find('input').text()).toEqual('');

    wrapper.instance().handleInput(event);

    expect(wrapper.state().display).toEqual('colorado');
  });

  it('sends the input text to searchDistrict method which will return an array of district objects', () => {
    const event = {
      preventDefault() {},
      target: { value: 'colorado' }
    };
    wrapper.instance().handleInput(event);
    expect(wrapper.instance().props.searchDistrict(event.target.value)).toEqual(
      containerData
    );
  });

  it('user should be able to pass in a case insensitive query', () => {
    const event = {
      preventDefault() {},
      target: { value: 'cOlOraDo' }
    };
    wrapper.instance().handleInput(event);
    expect(wrapper.instance().props.searchDistrict(event.target.value)).toEqual(
      containerData
    );
  });

  it('has a button which clears the input text and then causes all districts to display once more', () => {
    expect(wrapper.state().display).toEqual('');

    wrapper.find('input').simulate('change', { target: { value: 'Bou' } });
    expect(wrapper.state().display).toEqual('Bou')
    wrapper.find('button').simulate('click');

    expect(wrapper.state().display).toEqual('');
  });

});
