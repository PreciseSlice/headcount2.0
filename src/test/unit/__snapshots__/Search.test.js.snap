// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should have a label, input field, and a button in a form element 1`] = `
<header>
  <h5>
    Click two Colorado school districts to compare their full-day kindergarten participation rates
  </h5>
  <form>
    <label
      htmlFor="search-input"
    >
      Search districts: 
    </label>
    <input
      id="search-input"
      onChange={[Function]}
      onKeyPress={[Function]}
      type="text"
    />
    <button
      className="clear-button"
      onClick={[Function]}
    >
      Start Over
    </button>
  </form>
</header>
`;
